streams/src/main/java/org/apache/kafka/streams/state/internals/TimeOrderedWindowStoreBuilder.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.streams.state.WindowBytesStoreSupplier;
+import org.apache.kafka.streams.state.WindowStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+public class TimeOrderedWindowStoreBuilder<K, V> extends AbstractStoreBuilder<K, V, WindowStore<K, V>> {
```

Comment:
I'm not sure about this one yet. I found this issue in the `RocksDBTimeOrderedWindowStoreTest` which tests the store can be restored from a changelog. The only different thing with this builder is the `maybeWrapLogging` that returns an instance of `ChangeLoggingTimeOrderedWindowBytesStore`.

I believe we need to allow the store to be restored, right? If I used the generic `ChangeLoggingWindowBytesStore`, then the restore fails because the generic uses the `WindowKeySchema`.

Reply:
I just removed the `ChangeLoggingTimeOrderedWindowBytesStore` class in this new PR, and pass the window key serializer to the `ChangeLoggingWindowBytesStore` to reuse it. But I think I still need the `TimeOrderedWindowStoreBuilder`.

Reply:
Makes sense. I realized this store cannot be removed now.